### PR TITLE
[AUD-1478] Adds support for optimistic disbursements of aggregate challenges

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Tracks the individually disbursed challenges so that they can be coalesced into the optimistic challenge and calculate the claimable amount accordingly.

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

Not that I can think of

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

Locally against stage, mocking the response of the backend

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

Tons of manual testing

### CODE REVIEW

File: src/common/store/pages/audio-rewards/slice.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ type RewardsUIState = {
   userChallengesOverrides: Partial<
     Record<ChallengeRewardID, Partial<UserChallenge>>
   >
+  disbursedChallenges: Partial<Record<ChallengeRewardID, string[]>>
```
Comment (piazzatron) [inline src/common/store/pages/audio-rewards/slice.ts:59]: nitty: might be nice to leave a comment / or type `Specifier = string` to know what type the array value is

**Review by piazzatron — APPROVED** [review]
🚢

File: src/common/store/pages/audio-rewards/slice.ts (hunk 1 above)
Comment (rickyrombo) [reply src/common/store/pages/audio-rewards/slice.ts]: totally agree. actually decided to type specifier everywhere we use it

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-optimistic-disbursement

**Pull request merged**